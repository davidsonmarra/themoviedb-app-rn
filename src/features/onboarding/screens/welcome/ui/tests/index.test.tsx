import React from 'react';
import WelcomeContainer from '..';
import {render, screen} from '@testing-library/react-native';

const containerInstance = (props: any) =>
  render(<WelcomeContainer {...props} />);

describe('WelcomeContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const container = containerInstance({}).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should call the onPressContinue correctly', () => {
    const onPressContinue = jest.fn();
    containerInstance({onPressContinue});
    const button = screen.getByTestId('button-continue');
    button.props.onClick();
    expect(onPressContinue).toHaveBeenCalled();
  });
});
