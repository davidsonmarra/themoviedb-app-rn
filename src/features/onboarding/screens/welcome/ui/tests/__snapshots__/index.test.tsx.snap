// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeContainer should render correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/poster-avengers.jpg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "justifyContent": "space-between",
        "paddingHorizontal": 16,
        "paddingVertical": 24,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "testUri": "../../../src/assets/logo-the-movie-db.png",
        }
      }
      style={
        {
          "height": 86.33,
          "marginHorizontal": "auto",
          "width": 120,
        }
      }
    />
    <View
      style={
        {
          "gap": 12,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Seja bem vindo!
      </Text>
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 16,
            },
            undefined,
          ]
        }
      >
        Este é um aplicativo de filmes, utilizando a API pública do TMDB.
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#01B4A4",
          "borderRadius": 12,
          "justifyContent": "space-between",
          "opacity": 1,
          "padding": 8,
          "width": "100%",
        }
      }
      testID="button-continue"
    >
      <Text
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "bold",
              "padding": 8,
            },
            undefined,
          ]
        }
      >
        Continuar
      </Text>
    </View>
  </View>
</View>
`;
